src/main/java/graphql/validation/ValidationError.java

```diff
@@ -70,13 +71,18 @@ public ValidationError(ValidationErrorType validationErrorType, List<SourceLocat
 
     private ValidationError(Builder builder) {
         this.validationErrorType = builder.validationErrorType;
+        this.description = builder.description;
         if (builder.sourceLocations != null) {
             this.locations.addAll(builder.sourceLocations);
         }
-        this.description = builder.description;
-        this.message = builder.description;
-        this.queryPath = builder.queryPath;
-        this.extensions = builder.extensions;
+
+        if (builder.queryPath != null) {
+            this.queryPath.addAll(builder.queryPath);
+        }
+
+        if (builder.extensions != null) {
+            this.extensions.putAll(builder.extensions);
```

Comment:
This is the only point I don't like because the `putAll` method does not maintain the order